react-native/react/actions/pinentry.js

```diff
@@ -1,22 +1,96 @@
+/* @flow */
+
 import * as Constants from '../constants/pinentry'
+import engine from '../engine'
+
+import type {GUIEntryFeatures, GUIEntryArg} from '../constants/types/flow-types'
+import type {NewPinentryAction, RegisterPinentryListenerAction} from '../constants/pinentry'
+
+import type {Dispatch} from '../constants/types/flux'
+
+// TODO: there has to be a better way.
+const uglySessionIDResponseMapper: {[key: number]: Function} = {}
```

Comment:
We could theoretically avoid this by just keeping this within the context of the onSubmit function / onCancel function we pass to the smart component